onset_maproom/calc.py

```diff
@@ -2,51 +2,198 @@
 
+# Date Reading functions
+
+
 def read_zarr_data(zarr_path):
-  zarr_data = xr.open_zarr(zarr_path)
-  return zarr_data
-
-#Onset Date function
-
-def onset_date(daily_rain, early_start_day, early_start_month, search_days, rainy_day, running_days, running_total, min_rainy_days, dry_days, dry_spell, time_coord="T"):
-  """Fonction reproducing Ingrid onsetDate function
-  http://iridl.ldeo.columbia.edu/dochelp/Documentation/details/index.html?func=onsetDate
-  with the exception that:
-  output is a random deltatime rather than an actual onset
-  earlyStart input is now 2 arguments: day and month (as opposed to 1)
-  output is now the timedelta with each year's earlyStart (as opposed to the date itself)
-  """
-  onset_date = daily_rain[
-    (daily_rain[time_coord].dt.day==early_start_day)
-    &
-    (daily_rain[time_coord].dt.month==early_start_month)
-  ]
-  onset_date = xr.DataArray(
-    data=np.random.randint(0, high=search_days, size=onset_date.shape).astype('timedelta64[D]'),
-    dims=onset_date.dims,
-    coords=onset_date.coords,
-    attrs=dict(
-        description="Onset Date",
-    ),
-  )
-#Tip to get dates from timedelta  early_start_day
-#  onset_date = onset_date[time_coord] + onset_date  
-  return onset_date
+    zarr_data = xr.open_zarr(zarr_path)
+    return zarr_data
+
+
+# Growing season functions
+
+
+def onset_date(
+    daily_rain,
+    early_start_day,
+    early_start_month,
+    search_days,
+    rainy_day,
+    running_days,
+    running_total,
+    min_rainy_days,
+    dry_days,
+    dry_spell,
+    time_coord="T",
+):
+    """Function reproducing Ingrid onsetDate function
+    http://iridl.ldeo.columbia.edu/dochelp/Documentation/details/index.html?func=onsetDate
+    with the exception that:
+    output is a random deltatime rather than an actual onset
+    earlyStart input is now 2 arguments: day and month (as opposed to 1)
+    output is now the timedelta with each year's earlyStart (as opposed to the date itself)
+    """
+    onset_date = daily_rain[
+        (daily_rain[time_coord].dt.day == early_start_day)
+        & (daily_rain[time_coord].dt.month == early_start_month)
+    ]
+    onset_date = xr.DataArray(
+        data=np.random.randint(0, high=search_days, size=onset_date.shape).astype(
+            "timedelta64[D]"
+        ),
+        dims=onset_date.dims,
+        coords=onset_date.coords,
+        attrs=dict(
+            description="Onset Date",
+        ),
+    )
+    # Tip to get dates from timedelta  early_start_day
+    #  onset_date = onset_date[time_coord] + onset_date
+    return onset_date
+
+
+# Time functions
+
+
+def daily_tobegroupedby_season(
+    daily_data, start_day, start_month, end_day, end_month, time_coord="T"
+):
+    """Returns dataset ready to be grouped by with:
+    the daily data where all days not in season of interest are dropped
+    season_starts: an array where the non-dropped days are indexed by the first day of their season -- to use to groupby
```

Comment:
Today I learned that while black generally enforces a maximum line length of 88 characters, it makes an exception for docstrings. We have to keep lines like this short manually. Your editor might be able to fill paragraphs automatically.

Reply:
fixed manually. I am using vi on devi